llama_index/agent/openai_agent.py

```diff
@@ -53,6 +55,38 @@ def call_function(
     )
 
 
+class StreamingChatResponse:
+    """Streaming chat response to user and writing to chat history."""
+
+    def __init__(self, chat_stream: ChatResponseGen) -> None:
+        self.chat_stream = chat_stream
+        self.queue: Queue = Queue()
+        self.is_done = False
+        self.response = ""
+
+    def write_response_to_history(self, chat_history: List[ChatMessage]) -> None:
+        final_message = None
+        for chat in self.chat_stream:
+            final_message = chat.message
+            self.queue.put_nowait(chat.delta)
```

Comment:
Any reason in particular to use `put_nowait` instead of `put`? It prob doesn't make a difference right now given the `queue` we're using has an infinite size, but if we were to go forward with my above comment of optionally having the caller provide a queue instance, it may be better to just use the normal `put` and have it block when the queue is full.

Reply:
probably a good idea. I was just looking at some code that did something similar and the example also used `put_nowait` lol

Reply:
well, only a good idea if we go ahead with the above 🤔